addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -381,7 +381,7 @@ var SnippetEditor = Widget.extend({
                 this,
                 val.base_target ? this.$target.find(val.base_target).eq(0) : this.$target,
                 this.$el,
-                val.data,
+                _.extend(val.data, {snippetName: this._getName()}),
```

Comment:
Two problems with this:
- `_.extend` extends in-place so here you are actually changing `val.data`. When we don't want that, we usally add a third parameter `{}` as first parameter.
- Not really a problem but if for whatever reason `val.data` contains a `snippetName` value, it should be that one which is used as the `_getName()` here acts as default

You can solve both problems by inverting the parameters.